core/src/main/java/greencity/webcontroller/ManagementTipsAndTricksController.java

```diff
@@ -149,4 +161,22 @@ public GenericResponseDto save(@Valid
         }
         return buildGenericResponseDto(bindingResult);
     }
+
+    /**
+     * Returns  management page with User rating statistics with filtered data.
+     *
+     * @param model                ModelAndView that will be configured and returned to user.
+     * @param tipsAndTricksViewDto used for receive parameters for filters from UI.
+     */
+    @PostMapping(value = "", consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
+    public String filterData(Model model,
+                             @PageableDefault(value = 20) @ApiIgnore Pageable pageable,
+                             TipsAndTricksViewDto tipsAndTricksViewDto) {
+        PageableDto<TipsAndTricksDtoResponse> pageableDto =
+            tipsAndTricksService.getFilteredDataForManagementByPage(pageable,
+                tipsAndTricksService.getSpecification(tipsAndTricksViewDto));
```

Comment:
Conversion from `tipsAndTricksViewDto` to `Specification` should happen in `TipsAndTricksServiceImpl`.

`Specification` (and its implementations) is something database layer specific, so only Service Implementations can access it.

In one of my project I practiced to put method for `Specification` creation as default method of a `Repo` interface.